### DISCUSSION TITLE
Encryption hop-by-hop or end-to-end?

### DISCUSSION THREAD

**Discussion by martinoc000** [yggdrasil-network/yggdrasil-go#1204] [General]
Hi, is Yggdrasil's traffic encrypted hop-by-hop and not end-to-end right? I'm having difficulties to understand this without having tried it yet. I'm a bit confused by some documents that I found online related to this theme.

**Comment by NikitaPuzyryov** [comment]
Traffic is always end-to-end encrypted. _Protocol traffic_ _may_ be encrypted hop-by-hop if hops employ either `quic` or `tls` peerings.

See also: [How is Yggdrasil secured?](https://yggdrasil-network.github.io/implementation.html#how-is-yggdrasil-secured) ([link for current revision of the page for posterity](https://github.com/yggdrasil-network/yggdrasil-network.github.io/blob/94193456ac599d3e2f23a5b9c587a855924b288a/implementation.md#how-is-yggdrasil-secured))

**Reply by martinoc000** [reply]
Ok thank you very much. In chase of a direct connection to the Internet through a last node gateway, the website should reconise, thanks the peer-to-peer connection, the gateway ip. Are there other sensitive data that could be catch? I'm mostly wondering if the gateway reveal just the path of the previous node or if a hacker could easily find the entire path arriving to the source of the traffic.
